### DISCUSSION TITLE
Add Security Headers to Next Image Component URLs when reaching out to CMS

### DISCUSSION THREAD

**Discussion by JamesSingleton** [vercel/next.js#32988] [Help]
I was running [Screaming Frog](https://www.screamingfrog.co.uk/seo-spider/) against my website [Redshirt Sports](https://www.redshirtsports.xyz/?utm_source=GitHub) and even though I have security headers added in my `next.config.js`, it looks like they are not added to the URLs Next creates for the Image Component. I noticed that this is with URLs reaching out to https://sanity.io, so maybe there is something there as it does not complain about images I point to under the `public/` directory.

## Next Config

```js
const { withPlausibleProxy } = require('next-plausible')

const securityHeaders = [
  {
    key: 'X-Frame-Options',
    value: 'SAMEORIGIN',
  },
  {
    key: 'X-Content-Type-Options',
    value: 'nosniff',
  },
  {
    key: 'X-XSS-Protection',
    value: '1; mode=block',
  },
  {
    key: 'Referrer-Policy',
    value: 'origin-when-cross-origin',
  },
]
/** @type {import('next').NextConfig} */
module.exports = withPlausibleProxy()({
  productionBrowserSourceMaps: true,
  swcMinify: true,
  reactStrictMode: true,
  images: {
    formats: ['image/avif', 'image/webp'],
    domains: ['cdn.sanity.io', 'images.unsplash.com'],
  },
  async headers() {
    return [
      {
        source: '/(.*)',
        headers: [
          {
            key: 'x-robots-tag',
            value: 'all',
          },
        ],
      },
      {
        source: '/(.*)',
        headers: securityHeaders,
      },
    ]
  },
})
```

However, the URL for an [image](https://www.redshirtsports.xyz/_next/image?url=https%3A%2F%2Fcdn.sanity.io%2Fimages%2F8pbt9f8w%2Fproduction%2Fd64a119ddc4cdd6fa25caec9b2f0d05c659eb825-620x457.jpg%3Frect%3D0%2C54%2C620%2C349%26w%3D1020%26h%3D574%26fit%3Dmax%26auto%3Dformat&w=3840&q=75)

**Comment by volos-sq** [comment]
I faced with the same problem. There is no way to add custom headers when fetching images to the server.
If I understand correctly source code it is not possible right now.
https://github.com/vercel/next.js/blob/e74061161f174550d856711939e9b2865e8165e7/packages/next/src/server/image-optimizer.ts#L584
